lib/src/app.dart

```diff
@@ -117,7 +118,7 @@ class App {
   String get id => realmCore.appGetId(this);
 
   /// Create an app with a particular [AppConfiguration]
-  App(AppConfiguration configuration) : this._(realmCore.getApp(configuration));
+  App(AppConfiguration configuration) : this._(realmCore.getApp(configuration, scheduler.handle));
```

Comment:
First, why do you pass a handle here. We previously agreed that we should not pass handles to `_RealmCore`. Secondly, why pass a handle to a singleton?

Reply:
The reason is realm_core should not know about the rest of the SDK and should use an explicit interface. This is how it designed from the start. Thus methods in realm_core are an extension to `realm.h` and not part of the SDK. Then we can redesign the SDK without touching the realm_core interface and we can replace the realm_core implementation in the future while keeping the interface intact. 

The reason to pass handle here is since I did not want to include scheduler.dart in realm_core which gives very easy access to the singleton value and someone might use that in the future.

Reply:
`_RealmCore` is internal. It already knows how to read configs, handles, etc. We are not bound by its interface at all. I don't buy this argument.

Reply:
we have discussed this offline and I will be fixing it reverting these changes